packages/ingest/granule.js

```diff
@@ -56,13 +56,17 @@ class Discover {
     const file = _file;
     let test = new RegExp(this.collection.granuleIdExtraction);
     const match = file.name.match(test);
+    //console.log(match);
     if (match) {
-      granuleId = match[1];
+    
+      granuleId = match[0];
```

Comment:
@amarouane-ABDLHAK curious, why did you change this to `match[0]`

Reply:
We want the whole match of the file name to be the granuleId. match[1] will not work with a generic Regex expression for example if my regex is `^ghrc.*$` (all the files that start with keyword ghrc) and let say filename = 'ghrctest1.nc'
let test = new RegExp(`^ghrc.*$`);
match = filename.match(test); 
match[1] will be undefined.
![image](https://user-images.githubusercontent.com/33136280/34796897-b64db90a-f61c-11e7-8a46-17820d4ba87c.png)

Reply:
Your extraction regex must include a capturing group for this work: https://www.regular-expressions.info/refcapture.html

Reply:
Yes even with capturing the group match[0] will still give you the filename as the granuleId and prevent an error where somebody did not use include the capturing group in the regex. But I am glad it was fixed :)